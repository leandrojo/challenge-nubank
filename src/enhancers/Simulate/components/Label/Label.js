import styled from 'styled-components';

const Label = styled.label`
  color: ${p => p.theme.colors.grayDark};
  display: block;
  font-family: ${p => p.theme.fontFamily};
  font-size: 1.2rem;
  font-weight: 300;
  pointer-events: none;
`;

export default Label;
