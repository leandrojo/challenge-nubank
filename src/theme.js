const fontFamily = 'Rubik, sans-serif';

const theme = {
  backgroundColor: '#f5f5f5',
  colors: {
    primary: '#7f39b7',
    grayExtraDark: '#555',
    grayDark: '#767676',
    gray: '#999',
    grayLight: '#ccc',
    grayExtraLight: '#FAFAFA',
  },
  fontFamily,
  fontSize: '1em',
  typography: {
    heading: {
      color: '#252226',
    },
    paragraph: {
      color: '#716d73',
    },
  },
};

theme.components = {
  input: {
    css: {
      backgroundColor: 'transparent',
      fontFamily,
      fontSize: '1.4em',
      fontWeight: 300,
    },
  },
  pane: {
    backgroundColor: '#fff',
  },
  range: {
    emptyColor: theme.colors.grayLight,
  },
};

export default theme;
