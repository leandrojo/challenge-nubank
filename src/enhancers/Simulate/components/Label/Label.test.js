/* eslint-env jest */

import React from 'react';
import { ThemeProvider } from 'styled-components';
import renderer from 'react-test-renderer';

import theme from 'theme';

import Label from './Label';

describe('Simulate/Label', () => {
  it('with text', () => {
    const Component = () => (
      <ThemeProvider theme={theme}>
        <Label>First Deposit</Label>
      </ThemeProvider>
    );

    const expected = renderer.create(<Component />).toJSON();

    expect(expected).toMatchSnapshot();
  });
});
