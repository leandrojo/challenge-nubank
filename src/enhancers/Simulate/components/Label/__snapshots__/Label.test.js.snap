// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Simulate/Label with text 1`] = `
<label
  className="sc-bdVaJa gCSHtE"
>
  First Deposit
</label>
`;
